src/signer.rs

```diff
@@ -30,7 +30,7 @@ where
     BalanceOf<T>: From<u128> + Into<u128>,
 {
     /// Creates a new [`ScriptSignatureHandler`] with all blank signatures for the provided script.
-    pub(crate) fn new(script_args: &[Vec<u8>], signer_count: usize) -> Result<Self, Error<T>> {
+    pub(crate) fn new(script_args: &[&[u8]], signer_count: usize) -> Result<Self, Error<T>> {
```

Comment:
Last time I had to do with this whole script parameter serialization and these typical patterns around `ScriptTransaction` preparings - I had the idea, that a macro should be able to solve this (simplifying this common style, so that everybody has to write a little less copy&paste code). If you know, what I mean :blush: